include/deal.II/grid/grid_in.h

```diff
@@ -663,8 +663,26 @@ class GridIn
 
 
   DeclException0(ExcGmshNoCellInformation);
+  
+  /**
+   * Return a map containing field data. The format of the returned map is as follows: 
+   * - std::string stores the name of the field data (identifier) as specified in the external mesh
+   * - std::map<int, double > stores cell_id and value for the given identifier
+   */
+  std::map< std::string, std::map<int,double> > get_field_data() const
+  {      
+      return this->field_data;
+  }
 
 protected:
+    
+  /**
+  * Data member that stores cell data. The format is as follows:
+  * - std::string stores the name of the field data (identifier) as specified in the external mesh
+  * - std::map<int, double> stores cell_id and value for the given identifier
+  */
+  std::map< std::string, std::map<int,double> > field_data;
```

Comment:
This is the place where you store the map that I'm commenting on below. I think it would be easier to write this as `std::map< std::string, std::vector<double> >`.